pkg/controller/master/backup/backup.go

```diff
@@ -54,6 +54,7 @@ var vmBackupKind = harvesterv1.SchemeGroupVersion.WithKind(vmBackupKindName)
 // RegisterBackup register the vmBackup and volumeSnapshot controller
 func RegisterBackup(ctx context.Context, management *config.Management, opts config.Options) error {
```

Comment:
> Case 7: Detached PVC can't be taken a snapshot. (Negative case)

should we mount longhorn volume before snapshot?
https://github.com/harvester/harvester/pull/2648/commits/69342e8064589431cf3dde6f491bfbb43e7bc543

> Case 8: Cannot delete VM with disks if there is a snapshot for the disk. (Negative case)

Since longhorn snapshot cannot be restored without source PVC.
If user select delete disk when deleting a VM, it is more convenient for users to prompt and automatically delete the associated VM snapshot or PVC snapshot.

I have implemented delete related PVC snapshots when delete VM in PR https://github.com/harvester/harvester/pull/2648
I can add an enhancement to support delete VM snapshots automatically

Reply:
the volume snapshot PR also need enhancement: https://github.com/harvester/harvester/pull/2648#issuecomment-1231527983

Reply:
> should we mount longhorn volume before snapshot?

Yeah, I think it's good to mount it by ourselves, but do we want to wait for LH team to enhance it? https://github.com/longhorn/longhorn/issues/3726

Reply:
> In the other word, if a user chooses to retain a disk when deleting a VM, the disk and its snapshot will not be deleted. Other Volumes and VM snapshots and Volume snapshots are automatically deleted

If we automatically delete related VM snapshots and Volume snapshots, I think this behavior will surprise users.

cc @guangbochen @bk201 Do you prefer to use a webhook to stop this behavior, or automatically delete related resources? Thank you.

Reply:
> Yeah, I think it's good to mount it by ourselves, but do we want to wait for LH team to enhance it? https://github.com/longhorn/longhorn/issues/3726

https://github.com/harvester/harvester/blob/69a6b1b2517f30cd0dd81852cb575a29f0b1e683/pkg/controller/master/backup/backup.go#L148 
for old backup we also mounted the detached longhorn volume

Reply:
>  If we automatically delete related VM snapshots and Volume snapshots, I think this behavior will surprise users.

The snapshot that cannot used independently of the source volume is automatically deleted by most platforms, such as esxi and VirtualBox

Reply:
Hi @futuretea, I just updated my code and test cases. May you help me recheck it? Thank you.

Reply:
Updated test cases. Delete VMBackup snapshot when the source VM is deleted and also change VolumeSnapshot's owner reference when the VM is deleted. Please take a look again. Thanks.